enhancements/host-network-configuration.md

```diff
@@ -0,0 +1,484 @@
+        }
+      ]
+    },
+    "security": {
+      "tls": {
+        "certificateAuthorities": [
+          {
+            "source": "data:text/plain;charset=utf-8;base64,...==",
+            "verification": {}
+          }
+        ]
+      }
+    },
+    "timeouts": {},
+    "version": "2.2.0"
+  },
+  "networkd": {},
+  "passwd": {},
+  "storage": {},
+  "systemd": {}
+}
+```
+
+By customizing the “pointer” ignition configuration generated by the installer,
+additional network configuration can be done on the host prior to attempting to
+reach the Machine Config Server URL.
+
+#### Example
+
+Consider a cluster with hosts that have many network interfaces.  We may want
+to configure NetworkManager to not attempt to bring up interfaces that do not
+have explicit configuration.
+
+```
+mkdir lab
+cat << EOF > lab/etc/conf.d/10-disable-unconfigured-nics.conf
+[main]
+no-auto-default=*
+EOF
+```
+
+Next we update the two ignition configuration files generated by the installer
+to include our network configuration.   This example makes use of
+[filetranspiler](https://github.com/ashcrow/filetranspiler).
+
+```
+mv master.ign master.ign.orig
+mv worker.ign worker.ign.orig
+for i in master worker; do python3 ~/filetranspile --pretty -i ${i}.ign.orig -f ./lab -o ${i}.ign; done
+```
+
+### MachineConfig Resource
+
+Once a cluster is able to reach the Machine Config Server, an additional
+interface is available for managing host configuration: the `MachineConfig`
+resource.  The `MachineConfig` resource can be used to add additional host
+configuration for a single Node or pool of Nodes.  The OpenShift documentation
+provides some examples of using this interface, such as [this example of
+configuring an NTP server for the chrony time
+service](https://docs.openshift.com/container-platform/4.4/installing/install_config/installing-customizing.html#installation-special-config-crony_installing-customizing).
+
+This same interface can be used to manage custom NetworkManager configuration
+files.  As long as the network configuration is not needed during the initial
+bootstrapping of a new Node, this interface provides access to any host network
+configuration that’s doable through NetworkManager.
+
+#### Example
+
+Consider a bare metal cluster where the administrator would like to configure a
+VLAN interface after the cluster is up and running.  A `MachineConfig` resource
+can be used to apply a new NetworkManager configuration file to do this.
+
+First, we must craft a config file to drop into the
+`/etc/NetworkManager/system-connections `directory on each host. For example
+purposes, I used this config file, which configures a VLAN interface with a
+static IP instead of relying on DHCP to be present on this VLAN.
+
+In this case I used a VLAN ID of 20, and the NIC is ens3.
+
+```
+[connection]
+id=vlan-vlan20
+type=vlan
+interface-name=vlan20
+autoconnect=true
+[ipv4]
+method=manual
+addresses=172.5.0.2/24
+gateway=172.5.0.1
```

Comment:
I realized I should change this example back to using DHCP.  I had done a static config like this as a hack when testing, but it's not a realistic example since the `MachineConfig` interface applies to a pool of hosts and not a single host.

Reply:
updated